File: lib/models/user_model.dart
import 'package:client/services/config.dart';

class UserModel {
  String? id;
  String userName;
  String userId;
  String avatar;
  String biography = "";

  UserModel({
    required this.userName,
    this.userId = "",
    this.avatar = "$serverUrl/default.jpg",
    this.id,
  });
}

File: models/user-model.js
const {Schema, model} = require('mongoose')

const UserSchema = new Schema({
	email: {type: String, unique: true, required: true},
	password: {type: String, required: true},
	username: {type: String, required: true},
	avatar: {type: String, default: 'default.jpg'},
	biography: {type: String, default: null},
	creationDate: {type: String, default: Date.now()},
	followers: {type: Array, default: []},
	isActivated: {type: Boolean, default: false},
	activationLink: {type: String}
})

module.exports = model('User', UserSchema)